internal/docstore/firedocstore/fs.go

```diff
@@ -158,17 +160,44 @@ type collection struct {
 // firedocstore requires that a single string field, nameField, be designated the
 // primary key. Its values must be unique over all documents in the collection, and
 // the primary key must be provided to retrieve a document.
-func OpenCollection(client *vkit.Client, projectID, collPath, nameField string) *docstore.Collection {
-	return docstore.NewCollection(newCollection(client, projectID, collPath, nameField))
+func OpenCollection(client *vkit.Client, projectID, collPath, nameField string) (*docstore.Collection, error) {
+	c, err := newCollection(client, projectID, collPath, nameField, nil)
+	if err != nil {
+		return nil, err
+	}
+	return docstore.NewCollection(c), nil
+}
+
+// OpenCollectionFunc creates a *docstore.Collection representing a Firestore collection.
```

Comment:
Nit: I don't find this name descriptive. How about "OpenCollectionCompoundName" or "OpenCollection[With]NameFunc"?

Reply:
Went with OpenCollectionNameFunc.